client/lib/upgrades/actions/domain-management.js

```diff
@@ -296,6 +296,12 @@ function applyDnsTemplate( domainName, template, variables, onComplete ) {
 	} );
 }
 
+function getDnsTemplateConflicts( domainName, template, variables, onComplete ) {
```

Comment:
Ugh, I see I missed an important word in my above comment:
> For new code, I'm guessing we should **not** be adding new stuff here, instead of opting for doing it all the Redux Way (tm).

😅 But I think you got my point nevertheless 😁 As discussed on Slack, putting that call in the component itself (with some Promise magic) seems like a good way to go.